plugins/client-resources/src/index.ts

```diff
@@ -70,68 +79,123 @@ export default async () => {
+
+          const newOpt = { ...opt }
+          const connectTimeout = getMetadata(clientPlugin.metadata.ConnectionTimeout)
+          let connectPromise: Promise<void> | undefined
+          if ((connectTimeout ?? 0) > 0) {
+            connectPromise = new Promise<void>((resolve, reject) => {
+              const connectTO = setTimeout(() => {
+                if (!clientConnection.isConnected()) {
+                  newOpt.onConnect = undefined
+                  void clientConnection?.close()
+                  void opt?.onDialTimeout?.()
+                  reject(new Error(`Connection timeout, and no connection established to ${endpoint}`))
                 }
-              })
-            }
-            const clientConnection = connect(url, upgradeHandler, tokenPayload.workspace, tokenPayload.email, newOpt)
-            if (connectPromise !== undefined) {
-              await connectPromise
-            }
-            return await Promise.resolve(clientConnection)
-          },
-          filterModel ? [...getPlugins(), ...(getMetadata(clientPlugin.metadata.ExtraPlugins) ?? [])] : undefined,
-          createModelPersistence(getWSFromToken(token)),
-          opt?.ctx
-        )
+              }, connectTimeout)
+              newOpt.onConnect = (event) => {
+                // Any event is fine, it means server is alive.
+                clearTimeout(connectTO)
+                resolve()
+              }
+            })
+          }
+          const clientConnection = connect(url, upgradeHandler, tokenPayload.workspace, tokenPayload.email, newOpt)
+          if (connectPromise !== undefined) {
+            await connectPromise
+          }
+          return await Promise.resolve(clientConnection)
+        }
+
+        const modelFilter: ModelFilter = async (txes) => {
+          if (filterModel === 'client') {
+            return returnClientTxes(txes)
+          }
+          if (filterModel === 'ui') {
+            return returnUITxes(txes)
+          }
+          return txes
+        }
+
+        const client = createClient(handler, modelFilter, createModelPersistence(getWSFromToken(token)), opt?.ctx)
         return await client
       }
     }
   }
 }
+function returnUITxes (txes: Tx[]): Tx[] {
+  const configs = new Map<Ref<PluginConfiguration>, PluginConfiguration>()
+  fillConfiguration(txes, configs)
+
+  const allowedPlugins = [...getPlugins(), ...(getMetadata(clientPlugin.metadata.ExtraPlugins) ?? [])]
+  const excludedPlugins = Array.from(configs.values()).filter(
+    (it) => !it.enabled || !allowedPlugins.includes(it.pluginId)
+  )
+  return pluginFilterTx(excludedPlugins, configs, txes)
+}
+
+function returnClientTxes (txes: Tx[]): Tx[] {
+  const configs = new Map<Ref<PluginConfiguration>, PluginConfiguration>()
+  fillConfiguration(txes, configs)
+  const excludedPlugins = Array.from(configs.values()).filter((it) => !it.enabled || it.pluginId.startsWith('server-'))
+
+  const toExclude = new Set([
```

Comment:
if we add new class for UI specific tx, we will need to add new class here? Can we solve this with some attribute in tx instead?

Reply:
I've just filtered out most significant ones, so no need to update it often.